lib/bridge/QuitDebouncer.js

```diff
@@ -0,0 +1,125 @@
+/*eslint no-invalid-this: 0*/
+"use strict";
+var Promise = require("bluebird");
+
+function QuitDebouncer(ircBridge) {
+    this.ircBridge = ircBridge;
+
+    // Measure the probability of a net-split having just happened using QUIT frequency.
+    // This is to smooth incoming PART spam from IRC clients that suffer from a
+    // net-split (or other issues that lead to mass PART-ings)
+    this._debouncerForServer = {
+      // $server.domain: {
+            // $nick: {
+            //   promise: Promise, //Promise that resolves if the user joins a channel having quit
+            //   resolve: Function //function to call when a user joins
+            // }
+      // }
+    };
+
+    // Keep a track of the times at which debounceQuit was called, and use this to
+    // determine the rate at which quits are being received. This can then be used
+    // to detect net splits.
+    /*$server : [1477386173850, 1477386173825, ...]*/
+    this._quitTimesMs = {};
+    Object.keys(this.ircBridge.config.ircService.servers).forEach((domain) => {
+        this._quitTimesMs[domain] = [];
```

Comment:
Any reason why this isn't bundled with `_debouncerForServer`? They use the same keys after all, then you wouldn't need the null guard at `:37`.